src/components/Filters/index.tsx

```diff
@@ -0,0 +1,245 @@
+			if (!description) {
+				return;
+			}
+			setFormData({
+				field: description.field,
+				operator: 'contains',
+				value: description.value,
+			});
+			return;
+		}
+		setFormData(
+			editFilter.anyOf
+				?.map((f) => {
+					if (!isJSONSchema(f)) {
+						return;
+					}
+					const description = parseFilterDescription(f);
+					if (!description) {
+						return;
+					}
+					return {
+						field: description.field,
+						operator: description.operator,
+						value: description.value,
+					};
+				})
+				.filter((f) => !!f) as FormData[],
+		);
+	}, [editFilter, setFormData]);
+
+	const handleFilterChange = React.useCallback(
+		(filter: JSONSchema | null | undefined) => {
+			if (!filter) {
+				return;
+			}
+			let existingFilters = filters;
+			if (editFilter && filters) {
+				existingFilters = filters.filter((f) => f.$id !== editFilter.$id);
+			}
+			onFiltersChange?.([...(existingFilters ?? []), filter]);
+		},
+		[filters, onFiltersChange, editFilter],
+	);
+
+	const handleDeleteFilter = React.useCallback(
+		(filter: JSONSchema) => {
+			const newFilters = filters?.filter((f) => f.$id !== filter.$id) ?? [];
+			onFiltersChange?.(newFilters);
+		},
+		[filters, onFiltersChange],
+	);
+
+	const handleSaveView = React.useCallback(
+		(name: string) => {
+			const view: FiltersView = {
+				id: randomUUID(),
+				name,
+				eventName: `Saving filters view ${name}`,
+				filters: filters ?? [],
+			};
+			const newViews = [...(views ?? []), view];
+			onViewsChange?.(newViews);
+		},
+		[filters, views, onViewsChange],
+	);
+
+	const handleDeleteView = React.useCallback(
+		(view: FiltersView) => {
+			const newViews = views?.filter((v) => v.id !== view.id);
+			onViewsChange?.(newViews ?? []);
+		},
+		[views, onViewsChange],
+	);
+
+	return (
+		<Box>
+			<Box display="flex" alignItems="center" gap={2}>
+				{(!renderMode || renderMode.includes('search')) && (
+					<Box
+						ref={searchBarRef}
+						sx={{
+							display: 'flex',
+							flex: 1,
+							position: 'relative',
+							alignSelf: 'center',
+							zIndex: 10,
+						}}
+					>
+						<Search
+							id="balena__filters_search_bar"
+							focused={dark}
+							InputProps={{
+								style: { color: dark ? 'white' : 'primary' },
+							}}
+							onChange={(e) => {
+								if (!e) {
+									setSearchString('');
+									setShowSearchDropDown(false);
+									return;
+								}
+								setShowSearchDropDown(e.target.value !== '');
+								setSearchString(e.target.value);
+							}}
+							onEnter={(event) => {
+								event.preventDefault();
+								event.stopPropagation();
+								if (!searchString) {
+									return;
+								}
+								const filter = createFullTextSearchFilter(schema, searchString);
+								handleFilterChange(filter);
+								setSearchString('');
+							}}
+							value={searchString}
+						/>
+						{searchString && showSearchDropDown && onSearch?.(searchString)}
```

Comment:
Shouldn't this be inside the Search's onChange or onEnter (depending on how it was planned to be used, i.e. does the "Search" in onSearch refer to the search dropdown that appears while typing? Or does it refer to the act of searching by hitting enter?

Reply:
"onSearch" is the component you created :) It's the dropdown that appears while typing. I was really tempted to revise it, but I also thought that we will need to revisit it as soon as we need to implement it with server-side pagination logic. So, I thought it's better to leave it as it is for now.

Reply:
I know I created it, I just don't remember how it works anymore 😅 

We can revisit it later (if necessary since it's more pressing to explore AutoUI replacements as a whole), I'm just confused why I put the `onSearch` call directly into the render 🤔

Reply:
Ah my bad